src/LibraryManager.Vsix/Contracts/PerProjectLogger.cs

```diff
@@ -30,5 +33,78 @@ public void Log(string message, LogLevel level)
         {
             Logger.LogEvent($"{message}{ProjectName}", level);
         }
+
+
+        public void LogOperationSummary(IEnumerable<ILibraryInstallationResult> results, OperationType operation, TimeSpan elapsedTime)
+        {
+            int totalResultsCounts = results.Count();
+            IEnumerable<ILibraryInstallationResult> successfulRestores = results.Where(r => r.Success);
+            IEnumerable<ILibraryInstallationResult> failedRestores = results.Where(r => !r.Success);
+            IEnumerable<ILibraryInstallationResult> cancelledRestores = results.Where(r => r.Cancelled);
+            IEnumerable<ILibraryInstallationResult> upToDateRestores = results.Where(r => r.UpToDate);
+
+            bool allSuccess = successfulRestores.Count() == totalResultsCounts;
+            bool allFailed = failedRestores.Count() == totalResultsCounts;
+            bool allCancelled = cancelledRestores.Count() == totalResultsCounts;
+            bool allUpToDate = upToDateRestores.Count() == totalResultsCounts;
+            bool partialSuccess = successfulRestores.Count() < totalResultsCounts;
+
+            Logger.LogEvent("Restoring libraries completed", LogLevel.Status);
```

Comment:
hard coded strings here and below?

Reply:
Ignore these as well... My new PR has all fixed